### PR TITLE
fix(e2e): fix broken precompile e2e tests and enable CI

### PR DESCRIPTION
**Pull request by archbear**
erc20, gov and staking precompile tests still broken

add tests to ci

### CODE REVIEW

**Comment by calbera** [comment]
TODO: Disallow precompiles from accessing StateDB. Simple fix: remove `GetStateDB` from the `PrecompileEVM` interface.

File: cosmos/x/evm/plugins/txpool/mempool/mempool_reader.go [hunk 1]
```diff
@@ -153,6 +152,9 @@ func (etp *EthTxPool) Nonce(addr common.Address) uint64 {
 		}
 	}
 
+	// We move this here instead of defer as a slight optimization.
+	etp.mu.RUnlock()
```
Comment (itsdevbear) [inline cosmos/x/evm/plugins/txpool/mempool/mempool_reader.go]: this introduces a bug, use `defer()`

Comment (calbera) [reply cosmos/x/evm/plugins/txpool/mempool/mempool_reader.go]: Yea true it’s jank

**Review by ocnc — APPROVED** [review]

**Comment by itsdevbear** [comment]
lfg

**Pull request merged**